src/elements/unified-share-modal-element/UnifiedShareModalElement.js

```diff
@@ -0,0 +1,172 @@
+import { normalizeItemResponse, normalizeUserResponse } from './utils';
+import {
+    CLIENT_NAME_USM_ELEMENT,
+    FIELD_ENTERPRISE,
+    FIELD_HOSTNAME,
+    FIELD_ID,
+    FIELD_NAME,
+    FIELD_TYPE,
+    FIELD_SHARED_LINK,
+    FIELD_SHARED_LINK_FEATURES,
+    TYPE_FILE,
+    TYPE_FOLDER,
+    FIELD_PERMISSIONS,
+    FIELD_EXTENSION,
+    FIELD_DESCRIPTION,
+} from '../../constants';
+import type { ItemType } from '../../common/types/core';
+import type { item as itemFlowType } from '../../features/unified-share-modal/flowTypes';
+import type { USMElementItemAPIResponse, USMElementSharedLinkType } from './types';
+
+type USMElementProps = {
+    apiHost: string,
+    itemID: string,
+    itemType: ItemType,
+    language: string,
+    showFormOnly?: boolean,
+    token: string,
+};
+
+const elementMessages = defineMessages({
+    loadingError: {
+        defaultMessage: 'Could not load shared link for this item.',
+        description: 'Message that appears when the USM cannot be loaded.',
+        id: 'be.usm.loadingError',
+    },
+});
+
+function UnifiedShareModalElement(props: USMElementProps) {
+    const { apiHost, itemID, itemType, language, token }: USMElementProps = props;
+    const [item, setItem] = useState<?itemFlowType>(null);
+    const [sharedLink, setSharedLink] = useState<?USMElementSharedLinkType>(null);
+    const [currentUserID, setCurrentUserID] = useState<?string>(null);
+    const [errorExists, setErrorExists] = useState<boolean>(false);
+
+    const api = new API({
+        apiHost,
+        clientName: CLIENT_NAME_USM_ELEMENT,
+        id: `${itemType}_usm`,
+        token,
+    });
+
+    const resetState = useCallback(() => {
+        setItem(null);
+        setSharedLink(null);
+        setCurrentUserID(null);
+    }, [setItem, setSharedLink, setCurrentUserID]);
+
+    const getError = useCallback(() => {
+        setErrorExists(true);
+    }, [setErrorExists]);
+
+    const getUserSuccess = useCallback(
+        userData => {
+            const { id, userEnterpriseData } = normalizeUserResponse(userData);
+            setSharedLink({ ...sharedLink, ...userEnterpriseData });
+            setCurrentUserID(id);
+            setErrorExists(false);
+        },
+        [sharedLink],
+    );
+    const getItemSuccess = (itemData: USMElementItemAPIResponse) => {
+        const { item: itemFromAPI, sharedLink: sharedLinkFromAPI } = normalizeItemResponse(itemData);
+        setItem(itemFromAPI);
+        setSharedLink(sharedLinkFromAPI);
+        setErrorExists(false);
+    };
+
+    useEffect(() => {
+        resetState();
+    }, [resetState, token, itemID, itemType]);
+
+    useEffect(() => {
+        const getUserData = async () => {
+            await api.getUsersAPI(false).getUser(itemID, getUserSuccess, getError, {
+                fields: [FIELD_ENTERPRISE, FIELD_HOSTNAME],
+            });
+        };
+        if (item && sharedLink && !currentUserID) {
+            getUserData();
+        }
+    });
```

Comment:
More a question for my own knowledge, since I haven't used effects much: is there any recommended sorting order for useEffect calls? Just wondering since this one doesn't have skipping logic like the other two, maybe it should come first (since I'm equating this one to a typical call made in componentDidMount)?

Reply:
Good idea. I updated the order of the `useEffect` calls to reflect that the call to the Item API occurs before the call to the Users API.